tests/Api/Response/Parser/FiltersParserTest.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+namespace Findologic\Tests\Api\Response\Parser;
+
+use Findologic\Api\Response\Parser\FiltersParser;
+use Findologic\Api\Response\Response;
+use Findologic\Api\Services\ColorImage;
+use Plenty\Modules\Plugin\Libs\Contracts\LibraryCallContract;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Class FiltersParserTest
+ * @package Findologic\Tests\Api\Response\Parser
+ */
+class FiltersParserTest extends TestCase
+{
+    /**
+     * @var ColorImage|MockObject
+     */
+    protected $colorImage;
+
+    /**
+     * @var LibraryCallContract|MockObject
+     */
+    protected $libraryCallContract;
+
+    public function setUp()
+    {
+        $this->libraryCallContract = $this->getMockBuilder(LibraryCallContract::class)
+            ->disableOriginalConstructor()
+            ->setMethods([])
+            ->getMock();
+        $this->colorImage = $this->getMockBuilder(ColorImage::class)
+            ->disableOriginalConstructor()
+            ->setMethods([])
+            ->getMock();
+    }
+
+    public function testParse()
+    {
+        $responseMock = $this->getMockBuilder(Response::class)->disableOriginalConstructor()->setMethods(null)->getMock();
+        /** @var FiltersParser|MockObject $filtersParserMock */
+        $filtersParserMock = $this->getFiltersParserMock(['createResponseObject']);
+        $filtersParserMock->expects($this->any())->method('createResponseObject')->willReturn($responseMock);
+
+        $results = $filtersParserMock->parse($this->getNoFiltersResponse());
+        $this->assertEmpty($results);
```

Comment:
I would propose that you either extract the case with no filters in a separate test or you use a data provider to make it clear that this test covers multiple cases.